### PR TITLE
DTLS renegotiation

### PR DESCRIPTION
**Pull request by piotr-now**
## Description
This pull request contains tests for renegotiation

## Status
**READY** - for review; **HOLD** - for merge
This PR contains changes from another PR which is no merged yet. Please do review only the last commit

## Requires Backporting
 NO

### CODE REVIEW

File: tests/suites/test_suite_ssl.function [hunk 1]
```diff
@@ -1537,6 +1537,19 @@ exit:
     return ret;
 }
 
+/*
+ * Perform data exchanging between \p ssl_1 and \p ssl_2. Both of endpoints
+ * must be initialized and connected beforehand.
+ *
+ * \retval  0 on success, otherwise error code.
+ */
+int exchange_some_data( mbedtls_ssl_context *ssl_1,
```
Comment (AndrzejKurek) [inline tests/suites/test_suite_ssl.function]: I don't really like the `some` part of it, would rather recommend naming it "exchange_data", with the amount being a parameter, a hard-coded value in a particular test.

File: tests/suites/test_suite_ssl.function [hunk 2]
```diff
@@ -3564,3 +3577,90 @@ exit:
...
+                                           MBEDTLS_SSL_HANDSHAKE_OVER );
+    TEST_ASSERT( ret == 0 );
+    TEST_ASSERT( client.ssl.state == MBEDTLS_SSL_HANDSHAKE_OVER );
+    TEST_ASSERT( server.ssl.state == MBEDTLS_SSL_HANDSHAKE_OVER );
+
+    /* Start test with renegotiation */
+    TEST_ASSERT( server.ssl.renego_status == MBEDTLS_SSL_INITIAL_HANDSHAKE );
+    TEST_ASSERT( client.ssl.renego_status == MBEDTLS_SSL_INITIAL_HANDSHAKE );
+
+    /* After calling this function for the server, it only sends a handshake
```
Comment (AndrzejKurek) [inline tests/suites/test_suite_ssl.function]: I would suggest moving this, slightly edited, comment to line 3640, just before `exchange_some_data`, to clarify why exchanging some data is needed.

File: tests/suites/test_suite_ssl.function [hunk 3]
```diff
@@ -3564,3 +3577,90 @@ exit:
...
+    TEST_ASSERT( server.ssl.renego_status == MBEDTLS_SSL_RENEGOTIATION_PENDING );
+    TEST_ASSERT( client.ssl.renego_status == MBEDTLS_SSL_INITIAL_HANDSHAKE );
+
+    TEST_ASSERT( 0 == exchange_some_data( &(client.ssl), &(server.ssl) ) );
+    TEST_ASSERT( server.ssl.renego_status == MBEDTLS_SSL_RENEGOTIATION_DONE );
+    TEST_ASSERT( client.ssl.renego_status == MBEDTLS_SSL_RENEGOTIATION_DONE );
+
+    /* After calling mbedtls_ssl_renegotiate for the client all renegotiation
+     * should happen inside this function. However in this test, we cannot
+     * perform simultaneous communication betwen client and server so this
```
Comment (AndrzejKurek) [inline tests/suites/test_suite_ssl.function]:
```suggestion
     * perform simultaneous communication between client and server so this
```

File: tests/suites/test_suite_ssl.function [hunk 4]
```diff
@@ -3564,3 +3577,90 @@ exit:
...
+     * function will return waiting error on the socket. All rest of
```
Comment (AndrzejKurek) [inline tests/suites/test_suite_ssl.function]:
```suggestion
     * function will return waiting error on the socket. The rest of
```

**Comment by piotr-now** [comment]
I had to rebase it

**Pull request merged**